staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -978,6 +999,7 @@ func (r *Request) request(ctx context.Context, fn func(*http.Request, *http.Resp
 		if err != nil {
 			return err
 		}
+		req = req.WithContext(httptrace.WithClientTrace(req.Context(), newDNSMetricsTrace(ctx)))
```

Comment:
is this context piling up in each retry?

Reply:
good catch

Reply:
added check to make sure we only add context once